### ISSUE TITLE
 h_maxima() got an unexpected keyword argument 'footprint'

### ISSUE THREAD

**Issue by FiReTiTi** [vanvalenlab/deepcell-tf#670]
**Describe the bug**
I used DeepCell to segment nuclei (dapi stained) in an IF image.
I got the following error: TypeError: h_maxima() got an unexpected keyword argument 'footprint'

**To Reproduce**
I simply tried to execute: labels = app.predict(imstack, batch_size=1, image_mpp=None, compartment='nuclear')
imstack being my dapi image, with dapi duplicated to have two channels, see below.

image = skimage.io.imread(path) ;
imstack = numpy.stack((image, image), axis=-1)
imstack = numpy.expand_dims(imstack, 0)
labels = app.predict(imstack, batch_size=1, image_mpp=None, compartment='nuclear')

I've used Mesmer like that without any issues for quite some time until I updated it.

**Expected behavior**
Segmentation didn't work, execution failed.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: I got the same problem on both macOS Intel CPU and Centos with P100 GPU.
 - Version 0.12.5
 - Python Version 3.9

**Additional context**
Here is the full trace:
2023-06-22 23:47:53.933564: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
Traceback (most recent call last):
  File "/Users/me/Downloads/CyclicIF/Test - 2048x2048 - Segmentation/tmp/Script.py", line 15, in <module>
    labels = app.predict(imstack, batch_size=1, image_mpp=None, compartment='nuclear')
  File "/opt/miniconda3/envs/Python39/lib/python3.9/site-packages/deepcell/applications/mesmer.py", line 306, in predict
    return self._predict_segmentation(image,
  File "/opt/miniconda3/envs/Python39/lib/python3.9/site-packages/deepcell/applications/application.py", line 448, in _predict_segmentation
    label_image = self._postprocess(output_images, **postprocess_kwargs)
  File "/opt/miniconda3/envs/Python39/lib/python3.9/site-packages/deepcell/applications/application.py", line 215, in _postprocess
    image = self.postprocessing_fn(image, **kwargs)
  File "/opt/miniconda3/envs/Python39/lib/python3.9/site-packages/deepcell/applications/mesmer.py", line 142, in mesmer_postprocess
    label_images = deep_watershed(model_output['nuclear'],
  File "/opt/miniconda3/envs/Python39/lib/python3.9/site-packages/deepcell_toolbox/deep_watershed.py", line 179, in deep_watershed
    markers = h_maxima(image=maxima,
TypeError: h_maxima() got an unexpected keyword argument 'footprint'

**Comment by rossbar** [comment]
Which versions of `deepcell-toolbox` and `scikit-image` do you have installed in your environment? The `footprint` kwarg was added in skimage>0.18, so if you have a scikit-image version later than that, you shouldn't have problems.

**Comment by FiReTiTi** [comment]
I just pip installed deepcell without any specific constraint.
The versions that were installed are:
 - deepcell 0.12.5
 - deepcell-toolbox 0.12.0
 - scikit-image 0.18.3 and 0.19.3 (I tried on different environment).
I tried on Mac Intel, Mac M2, Linux Centos, same error.

**Comment by rossbar** [comment]
> I just pip installed deepcell without any specific constraint.

There may be problems if you're mixing `conda install` with `pip install` in a conda-managed environment. I'm fairly confident that this is related to the scikit-image version. You definitely should not see this error with scikit-image 0.19.X.

**Comment by FiReTiTi** [comment]
It's part of a bigger libraries. I just list Mesmer/Deepcell as a dependency.
I've used it that way for a long time (way more than a year) and I didn't have this issue.

Can I force DeepCell/scikit-image to be installed with conda despite the library being installed by pip?

Why would the pip install scikit-image 0.19.x version would be problematic if installed using pip? Why is it not the same?

**Comment by rossbar** [comment]
I did a deeper dive trying to figure out what's going on here and I think I've made some progress. It turns out there were some additional deprecated (and removed since scikit-image v0.20) keyword args that we were relying on. Usually we would expect to catch these via deprecation warnings and fix them; however, in this case we had these function calls nested inside of other functions which changed their level in the call stack, ultimately preventing the warnings from being raised in our test suite.

From a practical standpoint, this means the latest versions of `deepcell-tf` (0.12.5 and the recently released 0.12.6) will only work with scikit-image~=0.19.3. If the scikit-image version is <0.19, then you get error from the OP. If the scikit-image version is >=0.20.0, you will get other errors related to removed keyword arguments for internal calls to `binary_dilation`. The latter is pinned against in `setup.py`, but the former isn't.

**Comment by FiReTiTi** [comment]
Thank you for the explanations.
I guess this will be resolved in the 0.12.7 version. right?

If I'm right, ~=0.19.3 means >= 0.19.3 and < 0.20. I did a try using scikit-image=0.19.3 version, but I got the same error.

**Comment by rossbar** [comment]
> If I'm right, ~=0.19.3 means >= 0.19.3 and < 0.20.

It technically means any patch release in the 0.19.X series, so 0.19, 0.19.1, 0.19.2, etc. But yes the latest patch (currently 0.19.3) would be preferred.

> I did a try using scikit-image=0.19.3 version, but I got the same error.

I still maintain this is likely an environment issue - i.e. the environment you're running in does not actually have scikit-image 0.19.3 installed, but some older version. You can double-check this by referring to the [`scikit-image` v0.19 reference docs](https://scikit-image.org/docs/0.19.x/api/skimage.morphology.html?highlight=h_maxima#skimage.morphology.h_maxima), which shows that `footprint` is a valid keyword argument.

You could verify this by using the virtual environment system built-in to Python itself:

```bash
python -m venv dc-skimage-testenv
source dc-skimage-testenv/bin/activate  # ... or whichever script is appropriate for your preferred shell
pip list  # Verify the environment is empty - should only contain pip, setuptools, and maybe wheel
pip install deepcell==0.12.5 scikit-image==0.19.3
pip list  # Now you should see both deepcell 0.12.5 and scikit-image 0.19.3 installed in the environment
<your code which triggered the exception here>
```

Once #675 is in and released, it will no longer be possible to install deepcell with an older version of scikit-image. This *should* mean that the problem will "just go away"... however, if you rely on another package in the same environment which *requires* scikit-image<0.19, then it won't be possible to install those two packages in the same environment.
